src/diffusers/models/attention.py

```diff
@@ -245,6 +251,113 @@ def forward(self, hidden_states, encoder_hidden_states=None, timestep=None, retu
         return Transformer2DModelOutput(sample=output)
 
 
+class UnCLIPTransformerModel(ModelMixin, ConfigMixin):
```

Comment:
@patrickvonplaten ok with this being a separate model?

Reply:
Def agree!

Reply:
Could you maybe move this whole class into its own file? And I'd actually call it differently as Dalle2 introduced this design -> maybe actually call it "PriorTransformer" ? think many people have read the "prior" section of Dalle 2

Reply:
yessir